webpack.prod.js

```diff
@@ -0,0 +1,29 @@
+const { merge } = require('webpack-merge');
+// const CompressionPlugin = require('compression-webpack-plugin'); 
+const common = require('./webpack.common');
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+const CssMinimizerPlugin = require('css-minimizer-webpack-plugin'); 
+
+module.exports = merge(common, {
+  mode: 'production',
+  plugins: [
+    // new CompressionPlugin()
+    new MiniCssExtractPlugin({
+      filename: '[name].css', 
+      chunkFilename: '[id].css'
+    })
+  ],
+  module: {
+    rules: [
+      {
+        test: /\.css$/, 
+        use: [MiniCssExtractPlugin.loader, 'css-loader'],
+      }
+    ]
+  },
+  optimization: {
+    minimizer: [
+      new CssMinimizerPlugin()
+    ]
+  }
+});
```

Comment:
so to summarize, you split the old webpack.config into 3 files, a common one which will get merged with both of the other two. for the dev config you enabled the hot reload server and put it into development mode. and in the production mode thats not needed since it will not need to be rebuilt/hot served since it 'wont' be changing. also added some css optimizations plugins

Reply:
yeah I pretty much followed the docs here
https://webpack.js.org/guides/production/